articles/Frontend/Architecture/coding-guide.md

```diff
@@ -284,54 +279,136 @@ const testTaker = Object.assign(aUser, assignee);
+Later we call `delegate` to attach allowance operation from `person` to `allowance` module. Allowance methods then update person state
 
 4. Forwarding
 
-_TBD_
+Forwarding design pattern is used to completely forward data and control to other module. This is the good approach to manage shared states and distribute permission for shared state operation.
 
+```js
+const person1 = {
+  name: "Bob",
+};
+
+const person2 = {
+  name: "Alice",
+};
+
+const familyWallet = {
+  balance: 0,
+  earn(amount) {
+    this.balance += amount;
+  },
+  spend(amount) {
+    this.balance -= amount;
+  },
+  getBalance() {
+    return this.balance;
+  },
+};
+
+function forward(person, methods, provider) {
+  methods.forEach((methodName) => {
+    person[methodName] = function () {
+      return provider[methodName].apply(provider, arguments);
+    };
+  });
+}
+
+forward(person1, ["earn", "spend", "getBalance"], familyWallet);
+forward(person2, ["spend", "getBalance"], familyWallet);
+
+person1.earn(30);
+console.log(person1.getBalance()); // 30
+
+person2.spend(10);
+person1.spend(10);
+
+console.log(person2.getBalance()); // 10
+```
+
+`forward` function does late binding of `provider` mathods to person objects.
+Later we call `forward` to attach wallet operation from `person1` and `person2` to `familyWallet`.
 
 ### Factories
 
 > "The best thing about JavaScript is its implementation of functions. It got almost everything right. But, as you should expect with JavaScript, it didn't get everything right."
-*Douglas Crockford, JavaScript: The Good Parts*
+> _Douglas Crockford, JavaScript: The Good Parts_
 
 When a module needs to keep a state and hide some implementation details, the factory pattern will be selected.
 
 ```js
-var countDownFactory = function countDownFactory(config){
-    var currentValue = config.value || 0;   //private but accessible through the API
-    var interval = null;                    //kept private
-
-    return {
-        getValue : function getValue(){
-            //expose some internal va
-            return currentValue;
-        },
-        start: function start(){
-            interval = setInterval(function(){
-                currentValue--;
-            }, config.delay);
-        },
-        stop : function stop(){
-            clearInterval(interval);
-        },
-        reset : function reset(){
-
-        }
-    };
+var countDownFactory = function countDownFactory(config) {
+  var currentValue = config.value || 0; //private but accessible through the API
+  var interval = null; //kept private
+
+  return {
+    getValue: function getValue() {
+      //expose some internal va
+      return currentValue;
+    },
+    start: function start() {
+      interval = setInterval(function () {
+        currentValue--;
+      }, config.delay);
+    },
+    stop: function stop() {
+      clearInterval(interval);
+    },
+    reset: function reset() {},
+  };
```

Comment:
**chore:** For consistency, we should use the same ES notation for all examples, and if possible use the modern syntax. The other examples are written in ES6+, so better to apply the same here. And please keep the code style in sync too (4 spaces indent).

**suggestion:**
```suggestion
    return {
        getValue() {
            // expose some internal variable
            return currentValue;
        },
        get value() {
            // we can also use getter/setter notation to expose internal variables
            return currentValue;
        },
        start() {
            interval = setInterval(() => (currentValue--), config.delay);
        },
        stop() {
            clearInterval(interval);
        },
        reset() {},
    };
```